### ISSUE TITLE
Use RNG seed of MBD sim

### ISSUE THREAD

**Issue by richelbilderbeek** [richelbilderbeek/razzo#153]
After a `misc_params` is added to a `razzo_params` (`#151), the MBD tree simulation must use an RNG seed. 

```{r}
run_razzo <- function(
  razzo_params
) {
  check_razzo_params(razzo_params) # nolint raket function
  testit::assert(beastier::is_beast2_installed())

  # Simulate incipient species tree
  if (1 == 2) {
    # Set MBD tree generation seed, #153
    set.seed(razzo_params$misc_params$tree_sim_rng_seed)
  }
  # ...
}
```

Fix this Issue by remove the `if`` statement.

**Comment by richelbilderbeek** [comment]
I think Giovanni would like this one :+1:

**Comment by Giappo** [comment]
Will put RNG seed in `mbd_params`, instead of `misc_params` :rainbow:

**Comment by Giappo** [comment]
@richelbilderbeek what should be part of mbd_params?
possibilities are:
lambda, mu, nu, q, crown_age, cond, seed
I'll modify becosys::create_mdb_params accordingly in order to call it in razzo

**Comment by richelbilderbeek** [comment]
My ideas about this, feel free to ignore:

I'd vote for lambda, mu, nu and q being MBD parameters, as is the current situation. 
I'd vote for RNG seed to be in `misc_params`.
crown age is already stored by razzo in (the fiendlishly long) inference_params$mrca_prior$mrca_distr$mean$value
The conditioning can be something fixed in razzo, else a `misc_param`

:rainbow:

**Comment by Giappo** [comment]
Now seed is in mbd_params, we can still change it if you want.

create_full_parameters_files and create_test_parameters_files now generate seeds that go from 1:n_replicates. I hope this is what you wanted from the start. In case it's not lemme know. Give a look at the functions.

**Comment by richelbilderbeek** [comment]
Putting the seed in `mbd_params` is perfect. Using same seeds is fine now: I think every simulated tree phylogeny will be unique, as all combinations of MBD parameters and seed are unique. If I am incorrect, let me know :1st_place_medal:

**Comment by Giappo** [comment]
There is still the problem of the crown_age parameters, because this parameter is essential to simulate the tree. I refer to this problem also in #159.
TL;DR: I feel that mbd_params should contain all the parameters we need to simulate a tree (which is the first step of the pipeline), so: lambda, mu, nu, q, cond, crown_age and seed. The rest should probably follow. It looks just more logical to me.

**Comment by richelbilderbeek** [comment]
See #159: Spoiler: if you need it to go fast, just do it.
